src/main/java/org/sagebionetworks/web/client/jsinterop/DownloadConfirmationProps.java

```diff
@@ -0,0 +1,52 @@
+package org.sagebionetworks.web.client.jsinterop;
+
+import com.google.gwt.core.client.JavaScriptObject;
+import com.google.gwt.core.client.JsonUtils;
+
+import jsinterop.annotations.JsFunction;
+import jsinterop.annotations.JsNullable;
+import jsinterop.annotations.JsOverlay;
+import jsinterop.annotations.JsPackage;
+import jsinterop.annotations.JsType;
+
+@JsType(isNative = true, namespace = JsPackage.GLOBAL, name = "Object")
+public class DownloadConfirmationProps extends ReactComponentProps {
+	@FunctionalInterface
+	@JsFunction
+	public interface Callback {
+		void run();
+	}
+
+	@FunctionalInterface
+	@JsFunction
+	public interface GetQueryRequest {
+		JavaScriptObject run();
+	}
+
+	@JsNullable
+	GetQueryRequest getLastQueryRequest;
+	@JsNullable
+	String folderId;
+	@JsNullable
+	Callback fnClose;
+	@JsNullable
+	String downloadCartPageUrl;
+
+	@JsOverlay
+	public static DownloadConfirmationProps create(String queryBundleRequestJson, String folderId, Callback onClose) {
+		DownloadConfirmationProps props = new DownloadConfirmationProps();
+		props.downloadCartPageUrl = "/#!DownloadCart:0";
+		props.getLastQueryRequest = new GetQueryRequest() {
+			@Override
+			public JavaScriptObject run() {
+				if (queryBundleRequestJson != null)
+					return JsonUtils.safeEval(queryBundleRequestJson);
```

Comment:
This is where I avoided JsInterop, since it was a large rabbit hole to redefine all of the repo model objects JSInterop versions (query bundle request, query, sort items, ......).  So instead I return a "JavaScriptObject" that results from parsing the given json.